frontend/src/components/search/SearchResultsListItem.tsx

```diff
@@ -99,8 +102,11 @@ export default function SearchResultsListItem({
             <div className="grid-col tablet:order-3 overflow-hidden font-body-xs">
               <span className={metadataBorderClasses}>
                 <strong>Agency:</strong>{" "}
-                {opportunity?.summary?.agency_name
-                  ? opportunity?.summary?.agency_name
+                {opportunity?.summary?.agency_name &&
+                opportunity?.summary?.agency_code &&
+                agencyNameLookup
+                  ? // Use same exact label we're using for the agency filter list
+                    agencyNameLookup[opportunity?.summary?.agency_code]
```

Comment:
@andycochran - this is the key part here - make sure this is what you want. We're displaying the exact checkbox choice label (if it's actually in the `agencyNameLookup` )